index.js

```diff
@@ -1,60 +1,141 @@
+const cors = require("cors");
+const mongoose = require("mongoose");
+const PORT = 4000;
+const listingRoutes = express.Router();
 
-const routes = require("./routes")
-const db = require("./models")
+let Listing = require("./listing.model");
 
-const app = express();
 app.use(cors());
+app.use(bodyParser.json());
 
-// environment variable PORT or 3000 if unset
-const port = process.env.PORT || 3000;
+mongoose.connect("mongodb://127.0.0.1:27017/listings", {
+  useNewUrlParser: true,
+});
+const connection = mongoose.connection;
 
-// Add middleware for parsing the body to req.body
-// middlewares are executed in the order added, so add before routes
-app.use(express.json())
-app.use(express.urlencoded({ extended: true }))
-
-app.use((error, req, res, next) => {
-  if (res.headersSent) {
-    return next(err)
-  }
-  res.status(error.statusCode || error.status || 500).send({error: error })
-})
+connection.once("open", function () {
+  console.log("MongoDB database connection established successfully");
+});
 
-app.use((req, res, next) => {
-  req.models = db.models
-  next()
-})
-
-var options = {
-  explorer: true,
-  editor: true,
-  swaggerOptions: {
-    urls: [
-      {
-        url: '/swagger.yaml',
-        name: 'Spec1'
+//Get all Listings and find by Location
+listingRoutes.route("/").get(function (req, res) {
+  var name = req.query.name;
+  if (name) {
+    Listing.find({ Location: name }, function (err, listing) {
+      res.status(200).json(listing);
+    });
+  } else {
+    Listing.find(function (err, listings) {
+      if (err) {
+        console.log(err);
+      } else {
+        res.status(200).json(listings);
+        console.log();
       }
-    ]
+    });
   }
-}
+});
+
+//Create new Listing
+listingRoutes.route("/").post(function (req, res) {
+  let listing = new Listing(req.body);
+  listing
+    .save()
+    .then((listing) => {
+      res.status(201).json(listing);
+    })
+    .catch((err) => {
+      res.status(400).send(err);
+    });
+});
 
-app.use('/', express.static(__dirname + '/swagger'));
-app.use('/swagger', swaggerUi.serve, swaggerUi.setup(null, options));
-app.use('/', routes)
+//GET Listing by id
+listingRoutes.route("/:id").get(function (req, res, next) {
+  let id = req.params.id;
+  Listing.findById(id, function (err, listing) {
+    res.status(200).json(listing);
+  });
+});
 
+//DELETE Listing by id
+listingRoutes.route("/:id").delete((req, res) => {
+  Listing.findByIdAndDelete(req.params.id)
+    .then(() => res.status(200).send("Deleted"))
+    .catch((err) => res.status(400).json("Error: " + err));
+});
 
-// Start up the database, then the server and begin listen to requests
-if(process.env.NODE_ENV != "test") {
-  db.connectDb().then(() => {
-    const listener = app.listen(port, () => {
-      console.info(`Server is listening on port ${listener.address().port}.`);
-    })
-  }).catch((error) => {
-    console.error(error)
-  })
-}
+listingRoutes.route("/:id").put(function (req, res) {
```

Comment:
Kolla kodexempel 11 put and patch för hur du gör en put:

finns inte objektet, ska objektet skapas och status ska vara 201 (created)
finns objektet ska det ersättas och status ska vara 200 (ok)
är objektet detsamma ska status vara 204 (no content)

kolla kodexemplena så är det enkelt att göra likadant